src/main/java/bio/terra/service/auth/iam/IamResourceType.java

```diff
@@ -28,41 +32,37 @@ public String toString() {
     return samResourceName;
   }
 
+  @FunctionalInterface
+  private interface FilterFunction {
+    boolean apply(IamResourceType resourceType);
+  }
+
+  private static IamResourceType findIamResourceType(FilterFunction filter, String typeName) {
+    return Arrays.stream(IamResourceType.values())
```

Comment:
Since you're inside `IamResourceType` the class name is redundant here
```suggestion
    return Arrays.stream(values())
```